hibernate-core/src/main/java/org/hibernate/proxy/pojo/BasicLazyInitializer.java

```diff
@@ -89,6 +89,19 @@ else if ( method.equals( setIdentifierMethod ) ) {
 
 	}
 
+	Serializable getIdentifier(boolean initialize) {
+		if ( initialize ) {
+			initialize();
+		}
+		return getIdentifier();
```

Comment:
We might well improve code readability a little bit by changing the method parameter to 'needsToInitialize' or stuff like that, so causal code reader could  grasp the rough meaning solely based on method signature without digging into impl details in method body.

Not sure how to improve the method name of 'isInitializeProxyWhenAccessingIdentifier' though.
Feel free to ignore my comment. Not a big deal.

Reply:
Thanks @NathanQingyangXu , I renamed it `needsToInitializeProxy`